library/ajax/facility_ajax_code.php

```diff
@@ -17,15 +17,25 @@
 require_once("../../interface/globals.php");
 require_once("$srcdir/options.inc.php");
 
-if (!verifyCsrfToken($_POST["csrf_token_form"])) {
-    csrfNotVerified();
+if (empty($_POST) && empty($_GET)) {
+    die;
+}
+if (!empty($_POST)) {
+    if (!verifyCsrfToken($_POST["csrf_token_form"])) {
+        csrfNotVerified();
+    }
+}
+if (!empty($_GET)) {
+    if (!verifyCsrfToken($_GET["csrf_token_form"])) {
+        csrfNotVerified();
+    }
```

Comment:
@sjpadgett ,
hmmm, this is ugly. I am thinking of doing the following instead (and breaking the $_REQUEST rule :) ):
```
if (!verifyCsrfToken($_REQUEST["csrf_token_form"])) {
    csrfNotVerified();
}
```

Reply:
I see now how it makes for enforcing csrf more cumbersome, couldn't you do the validate in verifyCsrfToken?

Reply:
As I recall, there were use cases where needed to keep things like this. Note this has already evolved a bit after getting this into about half of the codebase so far.

Reply:
Changed to use above $_REQUEST, which should be just fine.